### ISSUE TITLE
return multiple variable from matlab to python

### ISSUE THREAD

**Issue by emqiong** [arokem/python-matlab-bridge#133]
It is easy to return one variable from Matlab using
>>res = mlab.run_func('path/to/jk.m', {'arg1': 3, 'arg2': 5})

How can I return more than one variable from Matlab. For example, my .m function is as below
function [lol, sos] = jk(args)
    arg1 = args.arg1;
    arg2 = args.arg2;
    lol = arg1 + arg2;
   sos = arg1 * arg2;
end

**Comment by arokem** [comment]
It's probably doable. I think that it would require changing the way that the matlab `feval` function is called in here: 

https://github.com/arokem/python-matlab-bridge/blob/master/pymatbridge/matlab/util/pymat_feval.m
https://github.com/arokem/python-matlab-bridge/blob/master/pymatbridge/matlab/util/run_dot_m.m

I am actually not sure: the problem is to make `feval` aware that you want a matrix of outputs, instead of just one. This is a very strange aspect of the Matlab design.

**Comment by blink1073** [comment]
We could use something like this:

`[resp{1:1}] = run_dot_m(func_path, arguments);`

Then the individual return values are available as `resp{1}`..`resp{n}`.  This defaults to returning the full `nargout` of a function.

**Comment by blink1073** [comment]
Actually, I was mistaken, we can pass `nargout` as an argument and this allows us to select `nargout`:

`[resp{1:nargout}] = run_dot_m(func_path, arguments);`

**Comment by blink1073** [comment]
Pretty slick actually, it even works for `nargout=0`.  See http://blogs.mathworks.com/loren/2009/04/14/convenient-nargout-behavior/ for where I got the idea.

**Comment by arokem** [comment]
This seems good for always capturing all of the outputs. What if the user
only wants the first output?

**Comment by blink1073** [comment]
Then they pass nargout=1.

**Comment by blink1073** [comment]
It works, I tried it.

**Comment by blink1073** [comment]
The current behaviour is nargout is always 1.

**Comment by arokem** [comment]
This: https://github.com/arokem/python-matlab-bridge/pull/135

Doesn't seem to quite cut it.

```python

>>> res = mlab.run_func('/Users/arokem/tmp/jk/jk.m', {'arg1': 3, 'arg2':
5})

>>> res

 {'message': 'Successfully completed request', 'result': [8], 'success':
'true'}

```

We somehow need to set nargin in advance?

**Comment by blink1073** [comment]
Right, it would be a kwarg in run_func. 



—

**Comment by blink1073** [comment]
I've got a mostly working version of this, I will send a PR against your PR later today.

**Comment by blink1073** [comment]
Closed by #135.
